tests/test_algorithm.py

```diff
@@ -1965,3 +1966,41 @@ def analyze(context, perf):
                                 analyze=analyze)
         results = algo.run(self.panel)
         self.assertIs(results, self.perf_ref)
+
+
+class TestRemoveData(TestCase):
+    """
+    tests if futures data is removed after expiry
+    """
+    def setUp(self):
+        dt = pd.Timestamp('2015-01-01', tz='UTC')
+        metadata = {0: {'asset_type': 'future', 'symbol': 'X',
+                        'expiration_date': dt + timedelta(days=4)},
+                    1: {'asset_type': 'future', 'symbol': 'Y',
+                        'expiration_date': dt + timedelta(days=7)}}
+
+        env = TradingEnvironment()
+        env.write_data(futures_data=metadata)
+
+        index_x = [dt, dt + timedelta(days=1), dt + timedelta(days=2),
+                   dt + timedelta(days=3), dt + timedelta(days=4)]
+        data_x = pd.DataFrame([[1, 100], [2, 100], [3, 100], [4, 100],
+                               [5, 100]],
+                              index=index_x, columns=['price', 'volume'])
+        index_y = [dt + timedelta(days=2), dt + timedelta(days=3),
+                   dt + timedelta(days=4), dt + timedelta(days=5),
+                   dt + timedelta(days=6)]
+        data_y = pd.DataFrame([[6, 100], [7, 100], [8, 100], [9, 100],
+                               [10, 100]],
+                              index=index_y, columns=['price', 'volume'])
+
+        pan = pd.Panel({0: data_x, 1: data_y})
+        self.source = DataPanelSource(pan)
+        self.algo = TestRemoveDataAlgo(env=env)
+
+    def test_remove_data(self):
+        self.algo.run(self.source)
+
+        expected_length = [1, 2, 2, 2, 2, 1]
+        for i, dt in enumerate(self.algo.data):
```

Comment:
'dt' as a variable name is a bit confusing just because we often use 'dt' to refer to datetimes/timestamps.